src/Kestrel.Core/Internal/Http/Http1Connection.cs

```diff
@@ -390,20 +406,18 @@ internal void EnsureHostHeaderExists()
 
                 // System.Uri doesn't not tell us if the port was in the original string or not.
                 // When IsDefaultPort = true, we will allow Host: with or without the default port
-                if (host != _absoluteRequestTarget.Authority)
+                if (hostText != _absoluteRequestTarget.Authority)
                 {
                     if (!_absoluteRequestTarget.IsDefaultPort
-                        || host != _absoluteRequestTarget.Authority + ":" + _absoluteRequestTarget.Port.ToString(CultureInfo.InvariantCulture))
+                        || hostText != _absoluteRequestTarget.Authority + ":" + _absoluteRequestTarget.Port.ToString(CultureInfo.InvariantCulture))
```

Comment:
@Tratcher This has nothing to do with this PR, but if the problem here is "System.Uri doesn't not tell us if the port was in the original string or not", why not just compare hostText to RawTarget instead of doing this complicated condition?

Reply:
HostText isn't expected to be the full value of RawTarget, only part of it.